### PR TITLE
PP-12444 cypress test for stripe details summary view

### PR DESCRIPTION
**Pull request by nlsteers**
## WHAT

- stripe details task summary view cypress test
- new middleware: enforcePaymentProviderType
  - restrict an express route to only be accessible to the configured payment provider

### CODE REVIEW

File: test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary-test-helpers.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const taskStatus = {
+  NOT_STARTED: 'Not yet started',
+  COMPLETE: 'Completed',
+  CANNOT_START: 'Cannot start yet'
```
Comment (oswaldquek) [inline test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary-test-helpers.js:4]: Do we need to test for "In progress"?

File: test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary.cy.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        cy.get('h1').should('contain', 'Stripe details')
+      })
+      describe('The settings navigation', () => {
+        it('should show stripe details', () => {
+          cy.get('.service-settings-nav')
+            .find('li')
+            .contains('Stripe details')
+            .then(li => {
+              cy.wrap(li)
+                .should('have.attr', 'href', `${SERVICE_SETTINGS_URL}/stripe-details`)
```
Comment (oswaldquek) [inline test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary.cy.js:62]: What part of the UI is this testing?

Comment (nlsteers) [reply test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary.cy.js:62]: It's testing that the stripe details link is visible in the service settings navigation

File: test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary-test-helpers.js (hunk 1 above)
Comment (nlsteers) [reply test/cypress/integration/simplified-account/service-settings/stripe-details/task-summary-test-helpers.js:4]: given that "in progress" is not implemented, we do not at this stage

**Review by oswaldquek — APPROVED** [review]
LGTM

**Pull request merged**